arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedParquetDefinitionLevelReader.java

```diff
@@ -415,7 +415,7 @@ protected void nextVal(
       int startOffset = ((BaseVariableWidthVector) vector).getStartOffset(idx);
       // It is possible that the data buffer was reallocated. So it is important to
       // not cache the data buffer reference but instead use vector.getDataBuffer().
-      vector.getDataBuffer().setBytes(startOffset, buffer.array(), buffer.position() + buffer.arrayOffset(),
+      vector.getDataBuffer().setBytes(startOffset, buffer, buffer.position(),
```

Comment:
I think it's even easier. Arrow has a `setBytes(long, ByteBuffer)` method that writes `bytes.remaining() = buffer.limit() - buffer.position()` bytes starting at `buffer.position()`. This should just be `vector.getDataBuffer().setBytes(startOffset, buffer)`.

Reply:
Fixed. Thanks!